src/app/wishlists/wishlist-fetch.ts

```diff
@@ -28,20 +42,27 @@ export function fetchWishList(): ThunkResult<Promise<void>> {
       existingWishLists?.wishListAndInfo?.wishListRolls?.length !==
       wishListAndInfo.wishListRolls.length
     ) {
-      dispatch(transformAndStoreWishList(wishListAndInfo));
+      dispatch(transformAndStoreWishList(wishListAndInfo, true));
     } else {
       console.log('Refreshed wishlist, but it matched the one we already have');
     }
   };
 }
 
 export function transformAndStoreWishList(
-  wishListAndInfo: WishListAndInfo
+  wishListAndInfo: WishListAndInfo,
+  wishListExternallySourced: boolean
 ): ThunkResult<Promise<void>> {
   return async (dispatch) => {
     if (wishListAndInfo.wishListRolls.length > 0) {
       dispatch(loadWishLists(wishListAndInfo));
 
+      if (wishListExternallySourced) {
+        dispatch(markWishListsFetched(new Date()));
+      } else {
+        dispatch(markWishListsFetched(undefined));
+      }
```

Comment:
Instead of dispatching another action, just mark the last load time in the reducer when you handle the `loadWishLists` action.